src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetRandomCommand.cs

```diff
@@ -520,23 +520,18 @@ protected override void EndProcessing()
                 int n = _chosenListItems.Count;
                 for (int i = 0; i < n; i++)
                 {
-                    // randomly choose an item to go into the i-th position
+                    // randomly choose j from [i...n)
                     int j = this.Generator.Next(i, n);
 
-                    // swap j-th item into i-th position
+                    // output the object at j
+                    this.WriteObject(_chosenListItems[j]);
+
+                    // move the i'th item to position j, except when i = j.
                     if (i != j)
                     {
-                        object tmp = _chosenListItems[i];
-                        _chosenListItems[i] = _chosenListItems[j];
-                        _chosenListItems[j] = tmp;
+                        _chosenListItems[j] = _chosenListItems[i];
```

Comment:
Doesn't this cause the value in `i` to be duplicated?  Shouldn't we swap like we did before?

Reply:
I understand now, the assignment effectively removes the output value from the list for the next iteration.  The comment should be updated to make this more clear.

Reply:
Will do it soon.

Reply:
Done